pkg/oci/layout/image.go

```diff
@@ -0,0 +1,101 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package layout
+
+import (
+	"fmt"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/layout"
+	"github.com/sigstore/cosign/pkg/oci"
+	"github.com/sigstore/cosign/pkg/oci/internal/signature"
+)
+
+// SignedImage provides access to a remote image reference, and its signatures.
+func SignedImage(path string) (oci.SignedImage, error) {
+	p, err := layout.FromPath(imagePath(path))
+	if err != nil {
+		return nil, err
+	}
+	// there should only be one image in the index, so we can pass in empty hash to get it
```

Comment:
This is ~fine, but I think aligning a little more closely with the OCI layout-isms would make things a bit nicer. E.g. if everything is reachable from `index.json`, I can just push it to a registry (with any tool that understands an OCI image layout) and have _every_ artifact be available in the registry.